### ISSUE TITLE
XFA use of CDATA in exData section causes unhandled exception

### ISSUE THREAD

**Issue by JDziurlaj** [mozilla/pdf.js#13500]
Attach (recommended) or Link to PDF file here:
[Acrobat.pdf](https://github.com/mozilla/pdf.js/files/6602628/Acrobat.pdf)

Configuration:
- Web browser and its version: Firefox 89.0 x64
- Operating system and its version: Windows 10.0.19043.985
- PDF.js version: pr/13479 at ec1dd06c
- Is a browser extension: No

Steps to reproduce the problem:
1. enable XFA support
2. Open Acrobat.pdf

What is the expected behavior? (add screenshot)
PDF renders 
![image](https://user-images.githubusercontent.com/7104454/120894068-92bce380-c5e4-11eb-89a8-767eb114d3a0.png)

What went wrong? (add screenshot)
Receive console error. PDF is not loaded
```
TypeError: this.exData[$content][$text] is not a function
```
This appears to be caused by the use of `CDATA` in `exData` tags containing XHTML content. For example:
```xml
<exData contentType="text/html"><![CDATA[<body xfa:APIVersion="2.7.0.0" xmlns="http://www.w3.org/1999/xhtml" xmlns:xfa="http://www.xfa.org/schema/xfa-data/1.0/"><p style="letter-spacing:0in;text-align:left"><span style="xfa-spacerun:yes;font-family:Arial Black;kerning-mode:pair;font-size:8.742pt;color:#e44b25"> </span></p><p style="letter-spacing:0in;text-align:left"><span style="font-family:Arial Black;kerning-mode:pair;font-size:8.742pt;color:#e44b25;xfa-spacerun:yes">CLIENT INFORMATION</span></p></body>]]></exData>
```
(Removal of CDATA wrapper in a simpler example made the error go away)
Link to a viewer (if hosted on a site other than mozilla.github.io/pdf.js or as Firefox/Chrome extension):

**Comment by marco-c** [comment]
The exception mentioned in the first comment is fixed now, but the PDF still fails to load with `TypeError: can't access property "size", this.font is null`.

**Comment by calixteman** [comment]
Yes, I'm aware of this issue, and I'm working a fix for some underlying issues (need to layout the text to compute some unknown dimensions).

**Comment by marco-c** [comment]
@JDziurlaj could you confirm everything is working as expected now?

**Comment by calixteman** [comment]
Some stuff is still broken in this pdf.
I'm working on it.
